### DISCUSSION TITLE
usePagination isLastPage的设置时机

### DISCUSSION THREAD

**Discussion by huchaomin** [alovajs/alova#337] [General]
usePagination hook 在倒数第二页向最后一页请求时， 最后一页请求还没完成 isLastPage 属性已经为true了。这样的设定是否合理呀，大家讨论一下

**Comment by MeetinaXD** [comment]
我个人的看法是，`isLastPage` 应该用在判断 是否允许/是否需要 加载下一页的场景。

举具体一个场景的例子：分页器有「上一页」和「下一页」按钮，当已经在最后一页时，下一页按钮不发送请求，否则`page.value += 1`
如果用户在倒数第二页向最后一页请求时，此时 `isLastPage`已经为true，那么在逻辑里就可以使用这个条件让按钮无动作。否则此时将`page.value += 1`请求不存在的数据页。

**Comment by panghujiajia** [comment]
文档是是这么说的：当前是否为最后一页，pageCount 有值时会通过 pageCount 和 page 对比得到，否则会通过列表数据长度是否少于 pagSize 得到。

当你请求最后一页，你传入的page是在倒数第二页的基础上+1，这时候page改变后，触发了isLastPage的判断，所以直接就是true了，从字面上理解，这个isLastPage也没毛病。

你可以尝试不给total值，这样isLastPage的判断变成通过实际数据长度来判断，应该可以解决你的问题。

**Reply by huchaomin** [reply]
嗯，知道其中的逻辑，就不会踩坑了

**Comment by huchaomin** [comment]
嗯，知道其中的逻辑

**Comment by huchaomin** [comment]
感谢大佬的指点
